File: UIViewController+ReportInappropriate.m
//
// Created by Taro Minowa on 4/15/14.
// Copyright (c) 2014 Higepon Taro Minowa. All rights reserved.
//

#import <MessageUI/MessageUI.h>
#import "UIViewController+ReportInappropriate.h"

@interface UIViewController () <MFMailComposeViewControllerDelegate>

@end

@implementation UIViewController (ReportInappropriate)

- (void)makeReportWithEmailAddress:(NSString *)emailAddress emailBody:(NSString *)emailBody
{
    MFMailComposeViewController *vc = [[MFMailComposeViewController alloc] init];
    [vc setMailComposeDelegate:self];
    [vc setSubject:[NSString stringWithFormat:@"%@ - Report Inappropriate", [self _getAppName]]];
    [vc setToRecipients:[[NSArray alloc] initWithObjects:emailAddress, nil]];
    [vc setMessageBody:emailBody isHTML:NO];
    [self presentViewController:vc animated:YES completion:nil];
}

- (void)mailComposeController:(MFMailComposeViewController *)controller
          didFinishWithResult:(MFMailComposeResult)result
                        error:(NSError *)error
{
    [self dismissViewControllerAnimated:YES completion:nil];
}

- (NSString *)_getAppName
{
    NSBundle *bundle = [NSBundle mainBundle];
    NSDictionary *info = [bundle infoDictionary];
    return [info objectForKey:@"CFBundleDisplayName"];
}

@end

File: UIViewController+ReportInappropriate.swift
//
// Created by Taro Minowa on 10/29/15.
// Copyright (c) 2015 Higepon Taro Minowa. All rights reserved.
//

import UIKit
import MessageUI

extension UIViewController: MFMailComposeViewControllerDelegate {

    func makeReport(withEmail email: String, messageBody: String) {
        if MFMailComposeViewController.canSendMail() {
            let mc: MFMailComposeViewController = MFMailComposeViewController()

            mc.mailComposeDelegate = self
            mc.setSubject("\(self.getAppName()) - Report Inappropriate")
            mc.setToRecipients([email])
            mc.setMessageBody(messageBody, isHTML: false)
            self.presentViewController(mc, animated: true, completion: nil)
        } else {

        }
    }

    private func getAppName() -> String {
        if let info = NSBundle.mainBundle().infoDictionary {
            if let appName = info["CFBundleName"] as? String {
                return appName
            }
        }
        return ""
    }

    public func mailComposeController(controller: MFMailComposeViewController, didFinishWithResult result: MFMailComposeResult, error: NSError?) {
        controller.dismissViewControllerAnimated(true, completion: nil)
    }
}
